batchflow/research/profiler.py

```diff
@@ -56,33 +59,34 @@ def show_profile_info(self, per_iter=False, per_experiment=False, detailed=False
                 result = result.sort_index(level=0)
 
         elif per_iter is False and detailed is True:
-            groupby = ['experiment', 'action', 'id'] if per_experiment else ['action', 'id']
+            groupby = ['experiment', 'unit', 'id'] if per_experiment else ['unit', 'id']
```

Comment:
I would prefer ` groupby = (['experiment']  if per_experiment else []) + ['unit', 'id']` to omit repeating the second term, but this is really up to you